src/MapViewController.swift

```diff
@@ -846,10 +867,13 @@ open class MapViewController: UIViewController, LocationManagerDelegate {
     let oldAnnotations = self.currentAnnotations
     self.currentAnnotations = Dictionary()
     for (annotation, marker) in oldAnnotations {
-      let newMarker = TGMarker.init(mapView: self.tgViewController)
+      let newMarker = Marker.init()
+      addMarker(newMarker)
+      //TODO: also set polyline, polygon etc
       newMarker.point = marker.point
-      newMarker.stylingString = marker.stylingString
-      self.currentAnnotations[annotation] = newMarker
+      newMarker.tgMarker.stylingString = marker.stylingString
+      newMarker.tgMarker.stylingPath = marker.stylingPath
```

Comment:
Updated to check if the path is empty and if not set it, otherwise, set the styling string since `Marker` objects will have one or the other set (depending on the constructor used)